extra/helpers.py

```diff
@@ -26,32 +24,28 @@ def early_exec(x):
   return early_exec
 
 def proc(itermaker, q) -> None:
-  for x in itermaker(): q.put(x)
-  q.put(None)
-  q.close()
+  try:
+    for x in itermaker(): q.put(x)
+  except Exception as e:
+    q.put(e)
+  finally:
+    q.put(None)
+    q.close()
 
 class _CloudpickleFunctionWrapper:
-  def __init__(self, fn):
-    self.fn = fn
-
-  def __getstate__(self):
-    return cloudpickle.dumps(self.fn)
-
-  def __setstate__(self, pfn):
-    self.fn = cloudpickle.loads(pfn)
-
-  def __call__(self, *args, **kwargs) -> Any:
-    return self.fn(*args, **kwargs)
+  def __init__(self, fn): self.fn = fn
+  def __getstate__(self): return cloudpickle.dumps(self.fn)
+  def __setstate__(self, pfn): self.fn = cloudpickle.loads(pfn)
+  def __call__(self, *args, **kwargs) -> Any:  return self.fn(*args, **kwargs)
 
 def cross_process(itermaker, maxsize=16):
   q: multiprocessing.Queue = multiprocessing.Queue(maxsize)
   # multiprocessing uses pickle which cannot dump lambdas, so use cloudpickle.
   p = multiprocessing.Process(target=proc, args=(_CloudpickleFunctionWrapper(itermaker), q))
-  #p.daemon = True
+  p.daemon = True
```

Comment:
I made it a daemon to prevent the multiprocessing module from trying to join the child process when the main process exits. Otherwise, in case of an exception happening in the child process, the main process tries to join it twice which raises an exception.
I think it's safe because the child process should always finish before the main process since the main process is waiting for the child process to send a `None` value before it exits the `cross_process` function.

Reply:
It's not a daemon because daemons can't spawn more processes